.github/workflows/build-deb.yml

```diff
@@ -0,0 +1,134 @@
+                      
+    - name: Install Conan
+      uses: turtlebrowser/get-conan@v1.2
+      
+    - name: Setup Conan
+      if: steps.cache.outputs.cache-hit != 'true'
+      run: |
+        conan profile detect -f
+      
+    - name: Cleanup Prior Build
+      if: steps.cache.outputs.cache-hit == 'true'
+      continue-on-error: true
+      run: |
+        conan remove mtconnect_agent -c
+
+    - name: Build and Test C++ Agent
+      shell: bash
+      run: |
+        export CTEST_OUTPUT_ON_FAILURE=TRUE
+        conan create . --build=missing -pr conan/profiles/gcc  -o shared=${{ matrix.shared }} -o cpack=True -o cpack_name=dist -o cpack_destination=${{ github.workspace }} --test-folder=
+    - name: Cleanse package version
+      run: |
+        if [[ $PACKAGE_VERSION == v*.*.*.* ]]; then
+          echo "PACKAGE_VERSION=${PACKAGE_VERSION:1}"" >> $GITHUB_ENV
+        elif [[ $PACKAGE_VERSION == [0-9].*.*.* ]]; then
+          echo "PACKAGE_VERSION=${PACKAGE_VERSION:0}"" >> $GITHUB_ENV
+        else
+          echo "PACKAGE_VERSION=0.0.0.0" >> $GITHUB_ENV
+        fi
+    - name: Prepare Debian Package
+      shell: bash
+      working-directory: ${{ github.workspace }}
+      run: |
+        mkdir -p pkgroot/home/edge/
+        ls -lah pkgroot
+        tar -xzf dist.tar.gz -C pkgroot/home/edge/
+        mv pkgroot/home/edge/dist pkgroot/home/edge/agent
+        mv pkgroot/home/edge/agent/share/mtconnect/* pkgroot/home/edge/agent/
+        rm -rf pkgroot/home/edge/agent/share
+        rm -rf pkgroot/home/edge/agent/docker
+        rm -rf pkgroot/home/edge/agent/demo
+    - name: Create Debian Package
+      id: create_debian_package
+      uses: jiro4989/build-deb-action@v3
+      with:
+        package: mtconnect-agent
+        package_root: ${{ github.workspace}}/pkgroot
+        version: ${{ env.PACKAGE_VERSION }}
+        arch: all
+        desc: "MTConnect Agent for Linux"
+        maintainer: Datanomix <support@datanomix.io>
+    - name: publish if release to packagecloud
+      if: ${{ env.PACKAGE_VERSION != '0.0.0.0' }}
+      uses: computology/packagecloud-github-action@v0.8
+      with:
+        package-name: ${{github.workspace}}/${{steps.create_debian_package.outputs.file_name}}
+        packagecloud-username: datanomix
+        packagecloud-reponame: dnx-edge
+        packagecloud-distro: ubuntu/focal
+        packagecloud-token: ${{ secrets.PACKAGE_CLOUD_API_WRITE_TOKEN }}
+
+    - name: Deploy packagecloud jammy deb
+      if: ${{ env.PACKAGE_VERSION != '0.0.0.0' }}
+      uses: computology/packagecloud-github-action@v0.8
+      with:
+        package-name: ${{github.workspace}}/${{steps.create_debian_package.outputs.file_name}}
+        packagecloud-username: datanomix
+        packagecloud-reponame: dnx-edge
+        packagecloud-distro: ubuntu/jammy
+        packagecloud-token: ${{ secrets.PACKAGE_CLOUD_API_WRITE_TOKEN }}
+
+    - name: Deploy packagecloud noble deb
```

Comment:
Is there any way to make this more automatic as we adopt new OSes and drop others?

Reply:
I've seen this done for RHEL to handle 6.X, 7.X, 8.X, but for Debian I think it was just one package

I think this is a package cloud ism